clients/instance/ibm-pi-cloud_connection_integration_test.go

```diff
@@ -0,0 +1,61 @@
+//go:build integration
+// +build integration
+
+package instance_test
+
+import (
+	"context"
+	"testing"
+
+	utl "internal/testutils"
+
+	client "github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCloudConnection(t *testing.T) {
+	utl.CloudConnectionPreCheck(t)
+
+	// create session and clients
+	session := utl.TestSession(t)
+	cloudConnectionClient := client.NewIBMPICloudConnectionClient(context.Background(), session, utl.CloudInstanceID)
+	jobClient := client.NewIBMPIJobClient(context.Background(), session, utl.CloudInstanceID)
+
+	// CREATE
+	body := &models.CloudConnectionCreate{
+		Name:  &utl.CloudConnectionName,
+		Speed: &utl.CloudConnectionSpeed,
+	}
+	createRespOk, createRespAccepted, err := cloudConnectionClient.Create(body)
+	require.Nil(t, err)
+
+	// get cloud connection id from recently created cloud connection
+	var CloudConnectionID string
+	if createRespOk != nil {
+		CloudConnectionID = *createRespOk.CloudConnectionID
+	} else {
+		CloudConnectionID = *createRespAccepted.CloudConnectionID
+		utl.WaitForJobState(t, jobClient, *createRespAccepted.JobRef.ID)
+	}
+	utl.TestMessage("CREATE Cloud Connection", CloudConnectionID, *createRespAccepted)
+
+	// GET
+	getResp, err := cloudConnectionClient.Get(CloudConnectionID)
+	require.Nil(t, err)
+	utl.TestMessage("GET Cloud Connection", CloudConnectionID, *getResp)
+	// verify variables match
+	require.Equal(t, *getResp.Name, utl.CloudConnectionName)
+	require.Equal(t, *getResp.Speed, utl.CloudConnectionSpeed)
+
+	// DELETE
+	delResp, err := cloudConnectionClient.Delete(CloudConnectionID)
```

Comment:
Can we write delete block in a func and defer it so that the resources are always destroyed.